elfloader-tool/include/arch-arm/cpuid.h

```diff
@@ -12,7 +12,7 @@
 uint32_t read_cpuid_id(void);
 
 /* read MP ID register from CPUID */
-uint32_t read_cpuid_mpidr(void);
```

Comment:
I wonder if we should drop the `cpuid.c` file completely and move all functions as `static inline` here. Seem they are just wrappers around some assembly instructions anyway.

Reply:
I think this can be in a general cleanup PR, same with the `autoconf.h` change